File: server/src/main/java/ru/mityugov/budget_planner/service/IncomeService.java
package ru.mityugov.budget_planner.service;

import ru.mityugov.budget_planner.exception.NotFoundException;
import ru.mityugov.budget_planner.service.dto.GetIncomeDto;
import ru.mityugov.budget_planner.service.dto.SaveIncomeDto;

public interface IncomeService {
    long saveIncome(SaveIncomeDto saveIncomeDto);

    GetIncomeDto getIncomeById(long id) throws NotFoundException;
}

File: src/main/kotlin/ru/mityugov/budgetplanner/service/income/IncomeService.kt
package ru.mityugov.budgetplanner.service.income

import ru.mityugov.budgetplanner.model.Income
import ru.mityugov.budgetplanner.service.dto.AddIncomeDto
import ru.mityugov.budgetplanner.service.income.dto.IncomeFullInfoDto
import java.time.LocalDate
import java.util.UUID

interface IncomeService {
    fun addIncome(dto: AddIncomeDto)
    fun getIncome(uuid: UUID): IncomeFullInfoDto
    fun getIncomes(startDate: LocalDate, endDate: LocalDate): List<Income>
}